resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/KubernetesClusterSchedulerBackend.scala

```diff
@@ -181,7 +184,51 @@ private[spark] class KubernetesClusterSchedulerBackend(
     super.getExecutorIds()
   }
 
+  private def labelLowPriorityExecs(execIds: Seq[String]) = {
+    // Only kick off the labeling task if we have a label.
+    conf.get(KUBERNETES_EXECUTOR_POD_DECOMMISSION_LABEL).foreach { label =>
+      val labelTask = new Runnable() {
+        override def run(): Unit = Utils.tryLogNonFatalError {
+
+          val podsToLabel = kubernetesClient.pods()
+            .withLabel(SPARK_APP_ID_LABEL, applicationId())
+            .withLabel(SPARK_ROLE_LABEL, SPARK_POD_EXECUTOR_ROLE)
+            .withLabelIn(SPARK_EXECUTOR_ID_LABEL, execIds: _*)
+            .list().getItems().asScala
+
+          podsToLabel.foreach { pod =>
+            kubernetesClient.pods()
+              .inNamespace(pod.getMetadata.getNamespace)
+              .withName(pod.getMetadata.getName)
+              .edit({p: Pod => new PodBuilder(p).editMetadata()
+                .addToLabels(label, conf.get(KUBERNETES_EXECUTOR_POD_DECOMMISSION_LABEL_VALUE))
+                .endMetadata()
+                .build()})
```

Comment:
I am a bit uncertain now (and no time to try this out) but do we need the Pod filtering twice? I mean `podsToLabel` already contains`Pod` instances. So is there a reason why the following lines are not enough?

```suggestion
            new PodBuilder(pod).editMetadata()
              .addToLabels(label, conf.get(KUBERNETES_EXECUTOR_POD_DECOMMISSION_LABEL_VALUE))
              .endMetadata()
              .build()
```

Reply:
I woke up during the night and looked into this a bit more and now I see  I was wrong with the suggestion and why.

The `new PodBuilder()... build()` will create a Pod model object (with the new metadata) but the changes should be communicated back to k8s API server for example via a PATCH request like it was done at

https://github.com/apache/spark/blob/5f41a2752fba617524c072ee4f0e3a4e5368be13/resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsLifecycleManager.scala#L178-L187

I like the ".edit(...)" method a bit better.